gslib/tests/test_hmac.py

```diff
@@ -357,3 +403,168 @@ def test_hmac_fails_for_xml(self):
+  def test_shim_translates_hmac_list_command(self, mock_list_hmac_keys):
+    fake_cloudsdk_dir = 'fake_dir'
+    project = 'test-project'
+    service_account = (
+        'test.service.account@test_project.iam.gserviceaccount.com')
+    command_expected_list_command_format = (([
+        'list', '-a', '-u', service_account, '-p', project
+    ], ('Gcloud Storage Command: {} alpha storage'
+        ' hmac list {} --all --service-account {} --project {}').format(
+            os.path.join('fake_dir', 'bin', 'gcloud'),
+            hmac._LIST_COMMAND_SHORT_FORMAT, service_account, project)), ([
+                'list', '-a', '-u', service_account, '-l', '-p', project
+            ], ('Gcloud Storage Command: {} alpha storage'
+                ' hmac list {} --all --service-account {} --long --project {}'
+               ).format(os.path.join('fake_dir', 'bin',
+                                     'gcloud'), hmac._DESCRIBE_COMMAND_FORMAT,
+                        service_account, project)))
+    hmac_key = apitools_messages.HmacKeyMetadata(
+        accessId='fake123456789',
+        projectId='test-project',
+        serviceAccountEmail=service_account,
+        state='ACTIVE',
+        timeCreated=datetime.datetime(2020, 1, 1, 00, 16, 00),
+        updated=datetime.datetime(2020, 1, 1, 00, 16, 00),
+        etag='ABCDEFGHIK=')
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        for (command, expected) in command_expected_list_command_format:
+          mock_list_hmac_keys.return_value = iter([hmac_key])
+          mock_log_handler = self.RunCommand('hmac',
+                                             args=command,
+                                             return_log_handler=True)
+          info_lines = '\n'.join(mock_log_handler.messages['info'])
+          self.assertIn(expected, info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'UpdateHmacKey',
+                     autospec=True)
+  def test_shim_translates_hmac_update_command(self, mock_update_hmac_key):
+    fake_cloudsdk_dir = 'fake_dir'
+    etag = 'ABCDEFGHIK='
+    project = 'test-project'
+    access_id = 'fake123456789'
+
+    option_expected_state_flag = (('INACTIVE', 'deactivate'), ('ACTIVE',
+                                                               'activate'))
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        for (option, expected_state_flag) in option_expected_state_flag:
```

Comment:
You probably want separate tests instead of putting a loop here. Not sure why you're mocking API responses, though.

Reply:
Done, I split this test into 2:

- `test_shim_translates_hmac_update_command_when_inactive_state_option_is_passed`
- `test_shim_translates_hmac_update_command_when_active_state_option_is_passed`